broker/machinebroker/server/event_list.go

```diff
@@ -55,10 +76,14 @@ func (s *Server) filterEvents(events []*iri.Event, filter *iri.EventFilter) []*i
 		sel = labels.SelectorFromSet(filter.LabelSelector)
 	)
 	for _, iriEvent := range events {
-		if !sel.Matches(labels.Set(iriEvent.Metadata.Labels)) {
+		if !sel.Matches(labels.Set(iriEvent.Spec.InvolvedObjectMeta.Labels)) {
 			continue
 		}
-		if (filter.EventsFromTime >= 0 && iriEvent.Spec.EventTime >= filter.EventsFromTime) && (filter.EventsToTime >= 0 && iriEvent.Spec.EventTime <= filter.EventsToTime) {
+		if filter.EventsFromTime > 0 && filter.EventsToTime > 0 {
+			if iriEvent.Spec.EventTime >= filter.EventsFromTime && iriEvent.Spec.EventTime <= filter.EventsToTime {
+				res = append(res, iriEvent)
+			}
+		} else {
 			res = append(res, iriEvent)
 		}
```

Comment:
Can we also streamline this that we only `append(res, iriEvent)`?

Reply:
done